File: Data Structures & Algorithms/majority-element/submission-1.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int ans = nums[0];
        int count = 1;
        for (const int& i:nums ){
            if (count==0){
                ans = i;
            }
            if (i != ans){
                count--;
            }
            if (i==ans){
                count++;
            }
        }
        return ans;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        counts = defaultdict(int)
        for i in nums:
            counts[i]+=1
        keys = list(counts.keys())
        vals = list(counts.values())
        ans = []
        for i in range(k):
            val = max(vals)
            ans.append(keys[vals.index(val)])
            keys.remove(keys[vals.index(val)])
            vals.remove(val)
        return ans

